STIPS/STIP-009.md

```diff
@@ -0,0 +1,743 @@
+
+
+#### OperatorAdded
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|operator| added operator address |
+
+
+#### OperatorRemoved
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|operator| removed operator address |
+
+#### AllowedAssetAdded
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|asset| added allowed asset |
+
+#### AllowedAssetRemoved
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|asset| added allowed asset |
+
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|ISetToken|setToken|Instance of SetToken|
+|address|factory|Address of factory contract used to deploy contract|
+|address|methodologist|Address of methodologist which serves as providing methodology for the index|
+|boolean|anyAssetAllowed|when true, assetAllowList restrictions are ignored |
+|mapping(address => ExtensionState)|extensionAllowList|Mapping to check if extension is enabled|
+|mapping(address => bool)|operatorAllowList|Mapping indicating if address is an approved operator|
+|mapping(address => bool)|assetAllowlist|Mapping indicating if asset is approved to be traded for, wrapped into, claimed, etc.|
+
+#### Private Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address[]|extensions|Array of enabled extensions|
+|address[]|operators|List of approved operators|
+|address[]|allowedAssets|Array of enabled extensions|
+
+#### Functions
+
+| Name  | Caller  | Description 	|
+|------	|------	|-------------	|
+|interactManager|extension|Interact with a module registered on the Set Token|
+|initializeExtension|extension|Initializes an added extension from PENDING to INITIALIZED state|
+|addExtensions|owner|Add a new extension that the DelegatedManager can call|
+|removeExtensions|owner|Remove an existing extension tracked by the DelegatedManager|
+|addOperators|owner|Add new operator(s) address|
+|addAllowedAssets|owner|Add new asset(s) that can be traded to, wrapped to, or claimed|
+|removeAllowedAssets|owner|Remove asset(s) so that it/they can't be traded to, wrapped to, or claimed|
+|setAnyAssetAllowed|owner|set anyAssetAllowed variable to true or false|
+|setMethodologist|owner|Update the methodologist address|
+|setManager|owner|Update the manager of the Set Token|
+|addModule|owner|Add module to Set Token|
+|removeModule|owner|Remove module from Set Token|
+
+#### Modifiers
+
+| Name  | Description   |
+|------ |-------------  |
+|onlyOwner| Requires that DelegatedManager `owner` is caller |
+|onlyExtension | Requires that msg.sender is an initialized extension in the `extensionAllowList` array |
+
+### Functions
+
+> constructor
+
+```solidity
+constructor(
+    ISetToken _setToken,
+    address _factory,
+    address _methodologist,
+    address[] memory _extensions,
+    address[] memory _operators,
+    address[] memory _allowedAssets
```

Comment:
I'm gonna add a `bool` here for `useAssetAllowlist`. In the factory if the array is empty it will be set to `false`, if it contains addresses we'll validate it includes all the assets currently in the Set and then pass in `true` for  `useAssetAllowlist`